### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Abelian.Ext

### PR DESCRIPTION
**Pull request by kim-em**
Apologies about the git history, a whole chain of PRs have been opened before the previous one was merged.

---

- [x] depends on: #4322

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4322
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by adamtopaz** [comment]
bors d+

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4322~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Abelian/Ext.lean [hunk 1]
```diff
@@ -38,47 +38,36 @@ open CategoryTheory
...
-        rw [(linear_yoneda R C).map_id, ← unop_id, nat_trans.right_op_id, nat_trans.left_derived_id]
-        rfl
-      map_comp' := by
-        intro X Y Z f g
-        rw [(linear_yoneda R C).map_comp, nat_trans.right_op_comp, nat_trans.left_derived_comp]
-        rfl }
-#align Ext ext
+      -- Porting note: if we use dot notation for any of
+      -- `NatTrans.leftOp` / `NatTrans.rightOp` / `NatTrans.leftDerived`
+      -- then `aesop_cat` can not discharged the `map_id` and `map_comp` goals.
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Abelian/Ext.lean:52]:
```suggestion
      -- then `aesop_cat` can not discharge the `map_id` and `map_comp` goals.
```

**Comment by kbuzzard** [comment]
This file has never had a green tick on github (previous green ticks were other PRs merged into this). The error is
```
 /- The `checkType` linter reports:
THE STATEMENTS OF THE FOLLOWING DECLARATIONS DO NOT TYPE-CHECK. -/
-- Mathlib.CategoryTheory.Abelian.Ext
#check Ext_map.{u_1, u_2, u_3} /- the statement doesn't type check. -/
#check Ext_obj.{u_1, u_2, u_3} /- the statement doesn't type check. -/
make: *** [GNUmakefile:16: lint] Error 1
```

**Comment by semorrison** [comment]
> This file has never had a green tick on github (previous green ticks were other PRs merged into this). The error is
> 
> ```
>  /- The `checkType` linter reports:
> THE STATEMENTS OF THE FOLLOWING DECLARATIONS DO NOT TYPE-CHECK. -/
> -- Mathlib.CategoryTheory.Abelian.Ext
> #check Ext_map.{u_1, u_2, u_3} /- the statement doesn't type check. -/
> #check Ext_obj.{u_1, u_2, u_3} /- the statement doesn't type check. -/
> make: *** [GNUmakefile:16: lint] Error 1
> ```

These are lemmas generated by `simps`, without much thought in the original writing of this file. I verified they aren't used in mathlib3, and removed them with a note.

**Comment by kbuzzard** [comment]
There have been some minor changes since this was both d+'s and merged, but I've just looked over everything and this is still good to go.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**